### DISCUSSION TITLE
Handling parameter uncertainty in  mHM

### DISCUSSION THREAD

**Discussion by bibek-ux** [mhm-ufz/mHM#108] [Q&A] [answered]
Hello mHM community

I am trying to model soil moisture for Karnali region of Nepal using mHM. Thus, I assume handling parameter uncertainty in my analysis rather than relying on "most optimized" parameter will be crucial  since soil moisture is on of the states for modeling streamflow. 

I noticed there is something about parameter uncertainity at the bottom of the mhm.nml file under "mcmc specific". 
I can get some help on how to use this and how does this work?

Thanks!
![image](https://github.com/user-attachments/assets/d51093a4-3faa-40c4-b274-1c8e6cb488dd)

**Comment by rohinikumar82** [comment, marked as answer]
It's nice to hear that you want to tackle parametric uncertainty within mHM. The MCMC-specific approach is under reproach & development stage. We have not touched that for a long time. Another straightforward approach to tackle parametric uncertainty in mHM is to run DDS with a low budget (say 500 or 1000 times) for many times (say 100 or 200) - get 100 best parameter sets, and run the model with 100 best parakeet sets and perform the corresponding uncertainty assessment.  This kind of uncertainty approximation is mentioned in the paper by: Tolson, Bryan A., and Christine A. Shoemaker. "Efficient prediction uncertainty approximation in the calibration of environmental simulation models." Water Resources Research 44.4 (2008).  We used such approach in mHM. See for example the following paper: Samaniego, Luis, Rohini Kumar, and Matthias Zink. "Implications of parameter uncertainty on soil moisture drought analysis in Germany." Journal of Hydrometeorology 14.1 (2013): 47-68. 

Good luck

Best,
Rohini Kumar

**Reply by bibek-ux** [reply]
Thank you for the reply @rohinikumar82. I think I will take this approach to calculate the 95PPU and see if the  observed discharge falls in that band of simulated discharges. A few more doubts, if I may ask.

1. Can mhm be used to handle uncertainty in observed streamflow discharges. I think in models like swat , the we can adhere for 10 % uncertainty in the observed discharges,(though I do not know if it is model specific)

2.Do I initialize starting values of parameter sets for each optimization run randomly or is there anything I need to have in mind to avoid local minima 

Thank you !!

**Reply by rohinikumar82** [reply]
1. Currently, within the mHM code: optimization part - we do not use uncertainty in observed discharge. you can do that but this could be outside the mHM optimization routine. For example, you can run the first mHM with several parameter sets. Save the runs and then analyze them outside of the mHM code in a R post-processing scripts.   

2.  Use different starting seeds (check-in mhm.nml -> DDS-related parameters). In this case, even if you start with the same default parameter - you will end up with a different optimised parameter set. 

Hope the above is helpful for your settings.

Best,
Rohini
